resources/sdk/purecloudjavascript/templates/configuration.mustache

```diff
@@ -0,0 +1,128 @@
+const ConfigParser = require('configparser');
+const os = require('os');
+const path = require('path');
+const fs = require('fs');
+
+import Logger from './logger.js';
+
+class Configuration {
+	/**
+	 * Singleton getter
+	 */
+	get instance() {
+		return Configuration.instance;
+	}
+
+	/**
+	 * Singleton setter
+	 */
+	set instance(value) {
+		Configuration.instance = value;
+	}
+
+	constructor() {
+		if (!Configuration.instance) {
+			Configuration.instance = this;
+		}
+
+		this.configPath = typeof window !== 'undefined' ? '' : path.join(os.homedir(), '.genesyscloudjavascript', 'config');
+		this.refresh_access_token = true;
+		this.refresh_token_wait_max = 10;
+		this.live_reload_config = true;
+		this.host;
+		this.environment;
+		this.basePath;
+		this.authUrl;
+		this.config;
+		this.logger = new Logger();
+		this.setEnvironment();
+		this.liveLoadConfig();
+	}
+
+	liveLoadConfig() {
+		// If in browser, don't read config file, use default values
+		if (typeof window !== 'undefined') {
+			this.configPath = '';
+			return;
+		}
+
+		if (!fs.existsSync(this.configPath)) {
+			console.error('No config file exists in the specified path.');
+			return;
+		}
+
+		this.updateConfigFromFile();
+
+		if (this.live_reload_config && this.live_reload_config === true) {
+			const watcher = fs.watch(this.configPath, (eventType, filename) => {
+				this.updateConfigFromFile();
+				if (!this.live_reload_config) {
+					watcher.close();
+				}
+			});
+		}
+	}
+
+	setConfigPath(path) {
+		if (path) {
+			this.configPath = path;
+		}
+	}
+
+	updateConfigFromFile() {
+		var configparser = new ConfigParser();
+
+		try {
+			configparser.read(this.configPath); // If no error catched, indicates it's INI format
```

Comment:
An INI format config file didn't work correctly for me; wasn't picking up values on startup and when I updated it as the application was running

Reply:
Can you provide an example of INI format config file that's not working for you? I just copied the one in the design doc and use that one to test, and seems to work on my end.

Reply:
I've used the following code to print out the config object every second from my client
setInterval(() => {
  console.log(client.config)
}, 1000)

I think I was wrong about it not picking it up on startup but it's still not picking it up when it's changed live